apps/application/flow/step_node/ai_chat_step_node/impl/base_chat_node.py

```diff
@@ -111,7 +111,7 @@ def execute(self, model_id, system, prompt, dialogue_number, history_chat_record
         question = self.generate_prompt_question(prompt)
         self.context['question'] = question.content
         system = self.workflow_manage.generate_prompt(system)
-        self.context['system'] = question.content
+        self.context['system'] = system
         message_list = self.generate_message_list(system, prompt, history_message)
         self.context['message_list'] = message_list
         if stream:
```

Comment:
The provided code snippet is generally clean and concise. However, there's one small issue that could be improved:

```python
# Potential issue: In the line where 'self.context['system']' is being overwritten,
# it might be more appropriate to set it based on the current context rather than always overwriting it with `question.content`.

# Optimization suggestion: Consider handling potential exceptions when trying to access or modify context variables.

```

This comment suggests modifying the last assignment to use the updated system content instead of directly setting `self.context['system']` to `question.content`, which can prevent unexpected behavior if the context contains other relevant information that needs to be preserved. Additionally, adding error handling for accessing/modifying context variables can make the code more robust.